modules/ocf/functions/get_host_fqdns.pp

```diff
@@ -0,0 +1,15 @@
+# Get an array of FQDNs that a host is responsible for, with an optional suffix
+# The '@' DNS A record needs to be handled separately since @.ocf.berkeley.edu
+# doesn't really make sense as a FQDN.
+#
+# For example:
+# If there's a host with hostname 'death' and DNS A records 'dev-vhost' and DNS
+# CNAME records 'www', then this function (with the suffix 'ocf.io' would
+# return ['death.ocf.io', 'dev-vhost.ocf.io', 'www.ocf.io']
+function ocf::get_host_fqdns(String $suffix = 'ocf.berkeley.edu') {
```

Comment:
What is our typename policy these days? Do we care to add a return type too?

Reply:
I didn't add a return type because it doesn't work with older versions of puppet, and overheat doesn't have puppet available past 4.8, although maybe it could get a package from buster or something?

I'd ultimately like to add more types to this repo since they seem reasonably useful, but I'm not sure if we've really talked about it at all.

Reply:
Wait, they [actually work with 4.8](https://puppet.com/docs/puppet/5.3/lang_write_functions_in_puppet.html#return-types), I thought it was with above 4.8, not 4.8 and above, so adding a return type to this sounds like a good idea.